### PR TITLE
[Doc] Update top nav bar styles

### LINKED ISSUE #41838: [DOCS Theme] Change the theme of the top navigation bar
### Description

1. Change the theme and styles of the top nav bar
2. Add a gradient to the background (gradient colors are different for light mode and dark mode)

Figma - https://www.figma.com/file/DjtgroLeO6BQSD6Si5dedw/Ray-docs-design-system?type=design&node-id=1%3A176&mode=design&t=vSJREt0q8rq72Tu8-1

### Link

_No response_

### PR DESCRIPTION
**Pull request by peytondmurray**
## Why are these changes needed?

This PR updates some styles of the top nav bar.

- Changed `--pst-color-on-background` globally
- Fixed an issue where persistent navbar items would have incorrect spacing on mobile
- Moved the search button into the persistent navbar section

## Related issue number

Closes https://github.com/ray-project/ray/issues/41838. Currently blocked by https://github.com/ray-project/ray/pull/42231 getting merged.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/navbar.yml
Comment (aslonnie) [inline doc/source/navbar.yml:4]: is "Usecase" really one word?

Comment (peytondmurray) [reply doc/source/navbar.yml:4]: No, it isn't. The design calls for it regardless - I even brought this up with @simran-2797 and confirmed that it needed to be this way, so I think we need to keep it.

Comment (peytondmurray) [reply doc/source/navbar.yml:4]: Just saw the comment above mine, will change it now :sweat_smile:

**Comment by simran-2797** [comment]
Screenshot 2024-01-22 at 3 59 29 PM
Search icon in dark mode is not visible.

**Comment by simran-2797** [comment]
Can we remove underlines when I hover over the nav links? 
There is around 40px of space between the ray logo and the nav links. Can we reduce it and make it same as the spacing between individual nav links?

**Comment by peytondmurray** [comment]
> Can we remove underlines when I hover over the nav links?
Sure.

> There is around 40px of space between the ray logo and the nav links. Can we reduce it and make it same as the spacing between individual nav links?

Sure. FYI the padding is actually 1em on either side of the nav links, so it scales with font size.

**Review by simran-2797 — APPROVED** [review]

**Comment by peytondmurray** [comment]
@aslonnie This is ready to merge now, I think.

**Comment by aslonnie** [comment]
it needs ray docs owner approvals; adding @angelinalg (and she also has merge right).

**Review by aslonnie — APPROVED** [review]

**Comment by angelinalg** [comment]
That's nice! Thanks for your work on this, @simran-2797 and @peytondmurray.

**Review by angelinalg — APPROVED** [review]

**Pull request merged**
